scout/ecm_prep.py

```diff
@@ -13531,37 +13557,49 @@ def main(opts: argparse.NameSpace):  # noqa: F821
     # Initialize a list to track which individual ECMs contribute to packages
     ctrb_ms_pkg_prep = []
     # Identify all previously prepared measure packages
-    meas_prepped_pkgs = [
-        mpkg for mpkg in meas_summary if "contributing_ECMs" in mpkg.keys()]
+    meas_prepped_pkgs = [mpkg for mpkg in meas_summary if "contributing_ECMs" in mpkg.keys()]
+    # Identify and filter packages whose ECMs are not all present in individual ECM set
+    ecm_pkg_filtered = False
+    if opts.ecm_packages is not None:
+        ecm_pkg_filtered = True
+    ecm_names = [meas["name"] for meas in meas_toprep_indiv]
```

Comment:
I believe the intention of this line is to obtain a list of ECM name strings corresponding to ECMs named in the config yml (such that with further steps, packages are restricted to those that only use ECMs that were explicitly included in the config yml). Should we allow ECMs already processed to be available to the package, even if they weren't included in the config yml, or would doing so mean that ECMs prepared with different options could be inadvertently mixed?

Reply:
I think it probably makes sense to include any ECMs we have available, both prepared and to be prepared. Without that, the optimization to skip previously prepared ECMs could result in unexpected behavior for packages.

There remains an implicit expectation that the prepared ECM data should be deleted when running with different/new/altered ECMs.